parsec/core/fs/workspacefs/workspacefs.py

```diff
@@ -216,6 +225,8 @@ def workspace_name(self) -> str:
         source = FsPath(source)
         destination = FsPath(destination)
         real_destination = destination
+        if _destinsrc(source, destination):
+            raise OSError(errno.EINVAL)
```

Comment:
This change would deserve its own commit right? Also a message argument like `"Cannot move a directory '%s' into itself '%s'"` would be nice.